### PR TITLE
Fixes anti-magic protection not nullifying damage from magical projectiles.

### PR DESCRIPTION
**Pull request by Potato-Masher**
:cl: Potato Masher
fix: Fixes anti-magic protection (like that gained from the Chaplain's nullrod) not properly nullifying incoming damage from some magical projectiles.
fix: Spellcards now respect anti-magic protection like other magical projectiles.
/:cl:

While investigating a similar issue with spellcards, I found that some projectiles like the fireball or the Spellblade's projectile would still deal damage/dismember/etc. while the unique effects like the fireball's explosion would still take effect on targets with a source of magic protection.

In short, the null rod is more consistent and less shit at defending against a wizard or other spell-slinging foe.

Another issue of note is that Holymelons apply this protection inside of bags, which technically makes them stronger than nullrods (which must be in hand/pocket to work).

### CODE REVIEW

**Comment by GuyonBroadway** [comment]
I thought that the nullrod nullified cult magic specifically and not wizard magic.


Unless it does the latter inconsistently in which case I need to screech at something.

File: code/modules/projectiles/projectile/magic.dm [hunk 1]
```diff
@@ -309,14 +317,15 @@
...
+	. = ..()
+	if(ismob(A))
+		var/mob/M = A
 		if(M.anti_magic_check())
-			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
-			qdel(src)
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [M]!</span>")
+			damage = 0
+			dismemberment = 0
 			return
```
Comment (XDTM) [inline code/modules/projectiles/projectile/magic.dm]: Wait, the early return is not preventing the damage?

**Comment by XDTM** [comment]
i think that letting the projectiles exist with 0 damage might be mechanically different from deleting them altogether, i'd look into preventing the damage proc if simply returning on prehit is not enough.

**Comment by Potato-Masher** [comment]
@GuyonBroadway Its supposed to have both holy and magic, but it turns out that it wasn't properly blocking the damage of some projectiles. Some even said that they were blocked, but it didn't actually do anything.

@XDTM From what testing I have done it works, the only reason I would see it failing would be some edge case that I haven't predicted.
CTF lasers do something like this as well, as in they set the damage to 150 on prehit to anyone with CTF armor on, so I assume this should work just fine.

**Comment by XDTM** [comment]
What i mean is stuff like reflection; they'd bounce back with no damage, instead of being absorbed.

**Comment by optimumtact** [comment]
Setting the damage to zero is not the right way to fix it, it's to prevent the attack chain running at the appropriate point.

**Comment by Potato-Masher** [comment]
I realized that, but I have no clue where the damage is even applied with projectiles. Anyone mind pointing me in the right direction?
